### ISSUE TITLE
interpolation blend colors even when fully transparent

### ISSUE THREAD

**Issue by Edu371** [opencv/opencv#25473]
### System Information

OpenCV python version: 4.9.0.80
Operating System / Platform: Windows 10
Python version: 3.11.2

### Detailed description

when using any type of interpolation the border matches background color, which is expected, the problem is that even when the background alpha is 0, it is still blending the others RGB channels

### Steps to reproduce

```import cv2
import numpy as np
import math
from PIL import Image, ImageDraw

height = 100
width = 500

image =  Image.new("RGBA", (width, height), (0, 0, 0, 0))
draw = ImageDraw.Draw(image)
draw.rounded_rectangle((5, 5, width-5, height-5), fill='#0000ff', radius=10)

image = np.array(image.getdata(), dtype=np.uint8).reshape(height, width, 4)

angleInDegrees = 4

h, w = image.shape[:2]
img_c = (w / 2, h / 2)

rot = cv2.getRotationMatrix2D(img_c, angleInDegrees, 1)

rad = math.radians(angleInDegrees)
sin = math.sin(rad)
cos = math.cos(rad)
b_w = math.ceil((h * abs(sin)) + (w * abs(cos)))
b_h = math.ceil((h * abs(cos)) + (w * abs(sin)))

rot[0, 2] += ((b_w / 2) - img_c[0])
rot[1, 2] += ((b_h / 2) - img_c[1])

outImg = cv2.warpAffine(image, rot, (b_w, b_h), flags=cv2.INTER_LINEAR)

out_pil = Image.fromarray(outImg)
bg =  Image.new("RGBA", out_pil.size, (255, 255, 255, 255))
bg.paste(out_pil, (0, 0), out_pil)
```
![black](https://github.com/opencv/opencv/assets/57723873/56884b74-c831-43fe-a8ea-9ba6fe253d10)
```
image =  Image.new("RGBA", (width, height), (255, 0, 0, 0))
```
![red](https://github.com/opencv/opencv/assets/57723873/f48d5824-cefe-4ce5-8cce-01bec1403c95)
```
image =  Image.new("RGBA", (width, height), (0, 255, 0, 0))
```
![green](https://github.com/opencv/opencv/assets/57723873/fb1eb694-1fec-44e4-9840-79a304bfb7cb)

zoom in the images to see

### Issue submission checklist

- [X] I report the issue, it's not a question
- [X] I checked the problem with documentation, FAQ, open issues, forum.opencv.org, Stack Overflow, etc and have not found any solution
- [X] I updated to the latest OpenCV version and the issue is still there
- [X] There is reproducer code and related data files (videos, images, onnx, etc)

**Comment by asmorkalov** [comment]
warpAffine applies interpolation to all channels, including alpha. It means, that alpha value on rectangle border is not strictly 0 or 255.

**Comment by Edu371** [comment]
yes, but the problem is that when the background color is (255, 0, 0, 0) it is interpolating to red even though alpha is 0.

when image background is RGBA (0, 0, 0, 0)
![black_zoomed](https://github.com/opencv/opencv/assets/57723873/a1008ef1-1046-4d6f-8e76-6a3de0de1b21)

when image background is RGBA (255, 0, 0, 0)
![red_zoomed](https://github.com/opencv/opencv/assets/57723873/d57afa8f-746d-4516-a046-a2fb2154caeb)

when image background is RGBA (0, 255, 0, 0)
![green_zoomed](https://github.com/opencv/opencv/assets/57723873/602cf580-d335-4118-a77c-4576c026546a)

background is fully transparent so it be (0, 0, 0, 0) or (255, 0, 0, 0) or (0, 255, 0, 0) should produce same results, the RGB channels should be ignored when alpha is 0

this wikipedia articles describes this problem
[https://en.wikipedia.org/wiki/Alpha_compositing#Comparison](https://en.wikipedia.org/wiki/Alpha_compositing#Comparison)
```The most significant advantage of premultiplied alpha is that it allows for correct blending, interpolation, and filtering. Ordinary interpolation without premultiplied alpha leads to RGB information leaking out of fully transparent (A=0) regions, even though this RGB information is ideally invisible. When interpolating or filtering images with abrupt borders between transparent and opaque regions, this can result in borders of colors that were not visible in the original image. Errors also occur in areas of semitransparency because the RGB components are not correctly weighted, giving incorrectly high weighting to the color of the more transparent (lower alpha) pixels.```

**Comment by Edu371** [comment]
@asmorkalov
i provided more info about this ussue please reconcider
